src/AwsS3Adapter.php

```diff
@@ -548,8 +551,16 @@ protected function upload($path, $body, Config $config)
         $options = $this->getOptionsFromConfig($config);
         $acl = isset($options['ACL']) ? $options['ACL'] : 'private';
 
-        if ( ! isset($options['ContentType']) && is_string($body)) {
-            $options['ContentType'] = Util::guessMimeType($path, $body);
+        if (!isset($options['ContentType'])) {
+            if (is_string($body)) {
+                $options['ContentType'] = Util::guessMimeType($path, $body);
+            }
+            if (empty($options['ContentType'])) {
+                $extension = pathinfo($path, PATHINFO_EXTENSION);
+                if ($extension) {
+                    $options['ContentType'] = MimeType::detectByFileExtension($extension) ?: 'text/plain';
```

Comment:
@deviantintegral PR thephpleague/flysystem#617 has been merged in. Will you also remove the default MIME type `text/plain`? Looks quite odd to assume a default MIME type of `text/plain`, doesn't it?